src/py/pyodide/http.py

```diff
@@ -270,6 +278,10 @@ async def unpack_archive(
         filename = self._url.rsplit("/", -1)[-1]
         unpack_buffer(buf, filename=filename, format=format, extract_dir=extract_dir)
 
+    def abort(self, reason: Any = None) -> None:
+        assert self.abort_controller is not None, "abort_controller is not set"
```

Comment:
If we are going to allow setting None for `abort_controller`, I think it is better to either raise exception or ignore silently instead of using assert here.